core/diagnostics.rs

```diff
@@ -0,0 +1,871 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+
+// Note that source_map_mappings requires 0-indexed line and column numbers but
+// V8 Exceptions are 1-indexed.
+
+// TODO: This currently only applies to uncaught exceptions. It would be nice to
+// also have source maps for situations like this:
+//   const err = new Error("Boo!");
+//   console.log(err.stack);
+// It would require calling into Rust from Error.prototype.prepareStackTrace.
+
+use serde_json;
+use serde_json::value::Value;
+use std::fmt;
+use std::str;
+
+const K_MESSAGE_LOG: i64 = (1 << 0);
+const K_MESSAGE_DEBUG: i64 = (1 << 1);
+const K_MESSAGE_INFO: i64 = (1 << 2);
+const K_MESSAGE_ERROR: i64 = (1 << 3);
+const K_MESSAGE_WARNING: i64 = (1 << 4);
+
+#[derive(Debug, PartialEq, Clone)]
+pub enum DenoDiagnosticCategory {
```

Comment:
For this symbol and others, can you remove the "Deno" prefix? So just "DiagnosticCategory" instead of "DenoDiagnosticCategory" ? (And "Diagnostic" instead of "DenoDiagnostic")